### PR TITLE
Add trajectory controller integration tests

### PR DESCRIPTION
**Pull request by stefanscherzinger**
This adds some basic integration tests for the joint-based cartesian_trajectory_controller.

### CODE REVIEW

File: cartesian_trajectory_controller/test/test_controller.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+        self.switch = rospy.ServiceProxy('/controller_manager/switch_controller', SwitchController)
+        try:
+            self.switch.wait_for_service(timeout.to_sec())
+        except rospy.exceptions.ROSException as err:
+            self.fail(
+                "Could not reach controller switch service. Msg: {}".format(err))
+
+    def test_normal_execution(self):
+        """ Test the basic functionality on a straight line """
```
Comment (stefanscherzinger) [inline cartesian_trajectory_controller/test/test_controller.py:52]:
```suggestion
        """ Test the basic functionality with moving in a vertical square """
```

File: cartesian_trajectory_controller/test/test_controller.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            JointTrajectoryPoint(positions=q_start, time_from_start=rospy.Duration(3.0))]
+        self.set_joints.send_goal(
+            start_joint_state)
+        self.set_joints.wait_for_result()
+
+        srv = SwitchControllerRequest()
+        srv.stop_controllers = ['joint_trajectory_controller']
+        srv.start_controllers = ['cartesian_trajectory_controller']
+        srv.strictness = SwitchControllerRequest.BEST_EFFORT
+        self.switch(srv)
```
Comment (fmauch) [inline cartesian_trajectory_controller/test/test_controller.py:118]: Wouldn't it make more sense to move this block to the function that uses the Cartesian controller? Assuming that this function is called before `move_square()` makes sense, but is not necessarily directly obvious for future developers.

**Review by fmauch — COMMENTED** [review]
Looking good overall. See my comments.

File: cartesian_trajectory_controller/test/test_controller.py (hunk 2 above)
Comment (stefanscherzinger) [reply cartesian_trajectory_controller/test/test_controller.py:118]: Yes, you might be right. It's now a separate call every time we use Cartesian control, see 2c384d3

**Review by fmauch — APPROVED** [review]
lgtm

**Pull request merged**